src/lib.rs

```diff
@@ -192,7 +192,7 @@ pub use cap::Cap;
 pub use ioctls::device::DeviceFd;
 pub use ioctls::system::Kvm;
 pub use ioctls::vcpu::{VcpuExit, VcpuFd};
-pub use ioctls::vm::{IoEventAddress, NoDatamatch, VmFd};
+pub use ioctls::vm::{new_vmfd, IoEventAddress, NoDatamatch, VmFd};
```

Comment:
Hmmm, the `new_vmfd` was not intended to be exported as part of the public interface. Checkout the comment: https://github.com/rust-vmm/kvm-ioctls/blob/8ead44e3af3f803d3a1c228755db54fc004aa20d/src/ioctls/vm.rs#L769

Any particular reason you need this function to be public?

Reply:
We want to restore the VmFd object on live upgrade destination side, otherwise we need to expose struct VmFd's private fields.

Reply:
This change should be added to a new commit in any case.

If you want to have a public constructor for VmFd, that you should add a public constructor in the VmFd structure implementation. When I first wrote this code this is what I was trying to avoid, hence I couldn't have a public constructor because VmFd would be a public structure hence the constructor will also be public. This is why I wrote the helper function `new_vmfd` so that the function can be public in the module, but not exported in the public interface.

Reply:
Hi Andreea, I have reimplemented it in another way:)